board/piksiv3/post_image.sh

```diff
@@ -1,47 +1,62 @@
-  cp $UBOOT_BUILD_DIR/u-boot.img $OUTPUT_DIR
-  cp $BINARIES_DIR/uImage.${CFG} $OUTPUT_DIR
+  $UBOOT_DEV_DIR/tools/mkimage                                                \
+  -A arm -T firmware -C none -O u-boot -n "FPGA bitstream"                    \
+  -d $FIRMWARE_DIR/piksi_fpga.bit                                             \
+  $OUTPUT_DIR/piksi_fpga.img
+
+  $UBOOT_DEV_DIR/tools/image_table_util                                       \
+  --append --print --print-images                                             \
+  --out $OUTPUT_DIR/PiksiMulti-DEV-$FILE_GIT_VERSION.bin                      \
+  --name "Piksi Buildroot DEV $BR_GIT_VERSION"                                \
+  --timestamp $(date +%s)                                                     \
+  --hardware v3_$HW_CONFIG                                                    \
+  --image $UBOOT_DEV_DIR/spl/u-boot-spl-dtb.img --image-type uboot-spl        \
+  --image $UBOOT_DEV_DIR/u-boot.img --image-type uboot                        \
+  --image $OUTPUT_DIR/piksi_fpga.img --image-type fpga
+
+  rm $OUTPUT_DIR/piksi_fpga.img
 }
 
 generate_prod() {
-  local HW=$1
-  local CFG=piksiv3_$HW
-  local UBOOT_BUILD_DIR=$UBOOT_DIR/build/${CFG}_prod
-  local OUTPUT_DIR=$BINARIES_DIR/${CFG}_prod
-  local BR_GIT_VERSION=$(git -C $BR2_EXTERNAL describe --tags --dirty --always --match 'v[0-9]\.[0-9]')
-  # remove the githash from the filename for useability (leave it in the header)
-  local FILE_GIT_VERSION=${BR_GIT_VERSION%%-g*} 
-  mkdir -p $OUTPUT_DIR
-  cp $UBOOT_BUILD_DIR/tpl/boot.bin $OUTPUT_DIR
-
-  $UBOOT_BUILD_DIR/tools/image_table_util                                     \
+  $UBOOT_PROD_DIR/tools/image_table_util                                      \
   --append --print --print-images                                             \
-  --out $OUTPUT_DIR/PiksiMulti-$FILE_GIT_VERSION.bin                           \
+  --out $OUTPUT_DIR/PiksiMulti-$FILE_GIT_VERSION.bin                          \
   --name "Piksi Buildroot $BR_GIT_VERSION"                                    \
   --timestamp $(date +%s)                                                     \
-  --hardware v3_$HW                                                           \
-  --image $UBOOT_BUILD_DIR/spl/u-boot-spl-dtb.img --image-type uboot-spl      \
-  --image $UBOOT_BUILD_DIR/u-boot.img --image-type uboot                      \
+  --hardware v3_$HW_CONFIG                                                    \
+  --image $UBOOT_PROD_DIR/spl/u-boot-spl-dtb.img --image-type uboot-spl       \
+  --image $UBOOT_PROD_DIR/u-boot.img --image-type uboot                       \
   --image $BINARIES_DIR/uImage.$CFG --image-type linux
 }
 
-generate_dev $HW_CONFIG
+mkdir -p $OUTPUT_DIR
+
+cp $UBOOT_PROD_DIR/tpl/boot.bin $OUTPUT_DIR
+cp $BINARIES_DIR/uImage.${CFG} $OUTPUT_DIR
+
+if [ -e $FIRMWARE_DIR/piksi_fpga.bit ]; then
+  generate_dev
+else
+  echo "*** NO FIRMWARE FILES FOUND, NOT BUILDING DEVELOPMENT IMAGE ***"
```

Comment:
We can simplify this logic now that we require firmware images for both prod and dev. Or if you want to be able to build dev with only an FPGA image then clarify this message to state `NO FPGA IMAGE FOUND`.